modular_citadel/code/game/turfs/cit_turfs.dm

```diff
@@ -1,24 +1,18 @@
-//Yes, hi. This is the file that handles Citadel's turf modifications.
+/turf/open/floor
+	var/dirtlevel
+	var/candirtify = TRUE
```

Comment:
You could move these two vars to `/turf/`, so you avoid the `istype(T, /turf/open/floor)` and just check `T.candirtify` on the dirt's init and just remove it on delete.
Not that this stuff is supposed to work outside of this subtype. anyhow.